app/controllers/api/site_configuration_controller.rb

```diff
@@ -16,7 +16,22 @@ def show
   def update
     authorize :site_configuration, :update?
     if @site_configuration.update(site_configuration_params)
-      render_success @site_configuration, "site_configuration.update_success"
+      if params[:site_configuration].key?(:inbox_enabled)
+        desired_state = ActiveModel::Type::Boolean.new.cast(params[:site_configuration][:inbox_enabled])
+        unless [true, false].include?(desired_state)
```

Comment:
This seems like an overly verbose and redundant way of handling something that rails should manage automatically.